cmake/black.cmake

```diff
@@ -17,7 +19,23 @@ if(BLACK)
     message(STATUS "BLACK auto formatter: ${BLACK}")
     # common target to concatenate all BLACK.py targets
     add_custom_target(${BLACK_TARGET})
+    add_custom_target(${BLACK_LINT_TARGET})
     set(BLACK_FOUND TRUE)
+
+    # run black on python files that are at the root directory
+    set(ROOT_DIR_FILES "${CMAKE_CURRENT_SOURCE_DIR}/setup.py")
+    add_custom_target(black.setup
+        COMMAND ${BLACK} "-l 79" "-v" ${ROOT_DIR_FILES}
```

Comment:
79 characters line are a bit short, what would you think of using 100 or 120 characters lines?

Reply:
I would be more than happy to do that but we need to check with the others too. I would go for 100 characters lines.

Reply:
100 is too long, but 88 (the default) works for me -- it's just about at the limit of what I can fit on split screens without wrapping.

Reply:
I don't know where the 88 values come from: Python PEP8 is 79, most coding guides are 80, Rust, Java on Android and a few other default to 100.

I've found this article defending the 80 chars per line limit for the same reason: you can fit two panels on a standard 1080p monitor: https://nickjanetakis.com/blog/80-characters-per-line-is-a-standard-worth-sticking-to-even-today

I typically use tabs instead of split panels so I am not affected by this issue; but it seems to be a very valid reason (in addition to the PEP8 standard 79) to stay at 79 😃

Reply:
The rationale for the 88-character default used in "black" is here: https://github.com/psf/black/blob/master/docs/the_black_code_style.md#line-length (I'd also recommend watching the talk linked under "90-ish").

I usually use tabs as well for working on separate files, but side-by-side splits can be very useful in some cases (like diffing versions of a file).

Reply:
So should the 88 default be applied ? I don't have any preferences on the question.

Reply:
I'm happy with 88 too, we can also ask for more feedback on Slack if you think this is necessary

Reply:
I changed the setting so that anyone can see the result. It feels much more reasonable to me.

Reply:
I would argue to take the same as for `c++`, which is currently at 80 characters (cpplint config). It makes the code look more homogeneous.

Reply:
Hmm, there's no reason _a priori_ to apply the same style rules to C++ and Python code. They're different languages with different syntax. In particular, C++ is a free-form language, which makes it much easier to keep lines to a certain length (you can put line breaks basically wherever you choose), whereas Python is much more restrictive in where line breaks can appear. If allowing slightly longer lines in Python gives us more readable code, then let's go for it.

Reply:
I completely agree. But the question was about an opinion, you got mine. And it is based on this:
My screen width, resolution, and editor space does not change with the language I am using. Therefore, practically speaking, the same spacial restrictions apply to different languages.

Where necessary, I am not opposing using longer lines - we had that option already previously also on the c++ side with the clang-format on/off and on the python side.